runner.go

```diff
@@ -82,6 +82,45 @@ func ExecutorOptions(opts ...ExecutorOption) RunOption {
 	}
 }
 
+// groupTaskAndFlagArgs groups all tasks with their flag arguments.
+// Notably, this function does not group flags passed to taskrunner itself.
+// Those flags are stored via the flags package and are handled separately.
+func (r *Runtime) groupTaskAndFlagArgs(args []string) map[string][]string {
+	flagArgsPerTask := map[string][]string{}
+
+	var currTaskName string
+	var currFlagsList []string
+	for _, arg := range args {
+		// Check task registry to figure out whether arg the is a task or a flag.
+		val, ok := r.registry.definitions[arg]
```

Comment:
do you/should you guard against a case where a user registers a flag that shadows the name of an existing registered task?

Reply:
Oo yes that's a good edge case; I'll ad a check!

Reply:
Done (in registry commit)